src/job.rs

```
@@ -955,33 +964,28 @@ fn connect_to_inbox(context: &Context, inbox: &Imap) -> libc::c_int {
     ret_connected
 }
 
-fn send_mdn(context: &Context, msg_id: u32) {
-    if let Ok(mut mimefactory) = MimeFactory::load_mdn(context, msg_id) {
-        if unsafe { mimefactory.render() }.is_ok() {
-            add_smtp_job(context, Action::SendMdn, &mut mimefactory);
-        }
-    }
+fn send_mdn(context: &Context, msg_id: u32) -> Result<(), Error> {
+    let mut mimefactory = MimeFactory::load_mdn(context, msg_id)?;
+    unsafe { mimefactory.render()? };
+    add_smtp_job(context, Action::SendMdn, &mut mimefactory)?;
+
+    Ok(())
 }
```

note that the replaced send_mdn func would silently swallowseveral  errors.